src/TcUnit.TestAdapter/Models/SystemService.cs

```diff
@@ -206,8 +221,11 @@ public bool DownloadFile(string localFile, string remoteFile, AdsDirectory stand
 
             long totalSize = 0;
             bool successWrite = true;
-            while (br.PeekChar() != -1)
+            // use ReadByte instead of PeakChar because PeakChar sometimes fails on non UTF-8 files
+            while (br.BaseStream.ReadByte() != -1)
```

Comment:
See https://stackoverflow.com/questions/22690193/binary-reader-peek-byte-in-c-sharp

I was running into an intermittent issue where I would receive an exception `System.ArgumentException: The output char buffer is too small to contain the decoded characters, encoding 'Unicode (UTF-8)'`. This change appears to fix it.